DataFormats/EcalDigi/interface/EcalDataFrame_Ph2.h

```diff
@@ -0,0 +1,60 @@
+#ifndef DIGIECAL_ECALDATAFRAME_PH2_H
+#define DIGIECAL_ECALDATAFRAME_PH2_H
+#include "DataFormats/EcalDigi/interface/EcalLiteDTUSample.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/Common/interface/DataFrame.h"
+#include "CondFormats/EcalObjects/interface/EcalConstants.h"
+
+
+//#define EcalMgpaBitwiseGain12 1
+//#define EcalMgpaBitwiseGain6 2
+//#define EcalMgpaBitwiseGain1 3
+//#define EcalMgpaBitwiseGain0 0
+
+//#define EcalLiteBitwiseGain1 1
+
+/** \class EcalDataFrame_Ph2
+      
+*/
+class EcalDataFrame_Ph2 {
+public:
+  EcalDataFrame_Ph2() {}
+  // EcalDataFrame_Ph2(DetId i) :  m_data(i) {}
+  EcalDataFrame_Ph2(edm::DataFrame const& iframe) : m_data(iframe) {}
+
+  virtual ~EcalDataFrame_Ph2() {}
+
+  DetId id() const { return m_data.id(); }
+
+  int size() const { return m_data.size(); }
+
+  EcalLiteDTUSample operator[](int i) const { return m_data[i]; }
+  EcalLiteDTUSample sample(int i) const { return m_data[i]; }
+
+  // get the leading sample (the first non saturated sample)
+  // starting from the fourth sample
+  // (it relies on the fact that the unpaker will discard fixed gain0 DataFrame)
+  // .. sample numbering: [0, 9]
+  // .. return -1 in case of no saturation
```

Comment:
This comment needs to be changed for phase2.
Maybe the algorithm itself as well.